### DISCUSSION TITLE
SceneViewer preview

### DISCUSSION THREAD

**Discussion by hybridherbst** [google/model-viewer#2830] [Q&A]
### Description

The Google AR/VR SceneViewer preview website (important to test if experiences work in SceneViewer and the only way to see actual warnings and error messages for that) seems to be down.

The console shows an error about some stuff being deprecated in Chrome (minus the slow network warning):

![image](https://user-images.githubusercontent.com/2693840/134402056-c9c2aa77-df35-4d5f-81dc-ee2b6d61faa0.png)

#### Live Demo
https://arvr.google.com/scene-viewer-preview

### Version

- model-viewer: vX.X.X

### Browser Affected

- [X] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [ ] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [X] Windows

### AR

- [ ] WebXR
- [X] SceneViewer
- [ ] QuickLook

**Comment by elalish** [comment]
FYI @tpsiaki. For what it's worth, we're  in the  process of deprecating this in favor of our  editor anyway. Using that to deploy to Scene Viewer for AR is actually a more accurate test anyway (since the web and native code stacks have some differences). If you find anything is missing from our editor, please let us know and  we'll be sure to add it!

**Comment by hybridherbst** [comment]
Thanks @elalish , I'm not sure what you're referring to in terms of the editor replacing that functionality.
SceneViewer has a lot of special behaviours (e.g. stops rendering when a certain vertex count is reached, complains about specific UV channels, breaks when specific GLB features are used, interprets animations differently, interprets lights differently) that have absolutely nothing to do with the capabilities of model-viewer and what the model-viewer editor shows.

So far, that SceneViewer preview has been the only way to actually check _why_ a specific model doesn't render in SceneViewer (but renders in model-viewer). Will you provide a "SceneViewer compatibility mode" in the editor? That does sound a lot of additional work. 
While I agree that deploying to SceneViewer is the only way to fully verify, it does not show error messages, it only provides unhelpful "end-user style" messages like "Looks like something is wrong with this model".

Case in point and why I just noticed that SceneViewer Preview is broken now:
http://amuse.vision/ar/Nymph/ renders fine in model-viewer, the file is a valid GLB according to the validator, yet SceneViewer can't display it. My go-to solution in these cases was to drop the model into SceneViewer Preview because in ~80% of cases it would indeed have a useful piece of information why that doesn't work. If you have a way to figure that out with the editor, or a way to see debug messages for SceneViewer on device, I'm all ears ;)

**Comment by tpsiaki** [comment]
@hybridherbst thanks for the report. As Emmett mentioned we're working toward a point where validation can be done with model-viewer and there's consistency in terms of glTF support between Scene Viewer and model-viewer.

The issue has to do the way the previewer is iframed, so you can work around it for now by using the version of the previewer that's served in the previewers iFrame here: https://scene-viewer-preview.appspot.com/

**Comment by hybridherbst** [comment]
@tpsiaki feature parity between SceneViewer and model-viewer would obviously be ideal, or SceneViewer using any known and testable rendering framework (bei it model-viewer/three or Filament or ...) 😁- really looking forward to that!

Thanks for that link, I wasn't aware of that existing - this works and immediately shows me 3 candidates for issues that could prevent this file in particular from working in SceneViewer (which is helpful even if the Preview pretends it should be working).

**Comment by tpsiaki** [comment]
@hybridherbst one other note on Scene Viewer's handling of models - the warnings that you see in the previewer have gotten somewhat stale and aren't always indicative of real problems in Scene Viewer. At the moment if your model loads and displays in that previewer (which it does seem to for me) then it's very likely that it will work in Scene Viewer on mobile, even if there are some warnings showing up.

Of course, some of the warnings (like file size, texture size) will result in longer load times and higher GPU memory use, which might result in failures or poor experiences for users with poor connections or very low end devices. But, we don't have hard enforced limits about these particular warnings on Scene Viewer.

**Comment by hybridherbst** [comment]
@tpsiaki I understand these points well since they are pain points in our projects :)
Would be curious if the above link (http://amuse.vision/ar/Nymph/) opens in SceneViewer for you? Definitely doesn't for me.
Is there a way to see debug messags from on-device SceneViewer on Android? I looked in adb but there weren't any as far as I remember.

**Comment by tpsiaki** [comment]
@hybridherbst it doesn't work for me but the reason it's failing is that you're passing this parameter for the sound file: `sound=http%3A%2F%2Famuse.vision%2Far%2FNymph%2F`. Scene Viewer is then trying to load http://amuse.vision/ar/Nymph/ as an audio file. If you remove the sound parameter, or set it to something that's a valid audio file it should work. For example, this link loads correctly for me (which is the same as in your demo, but with sound param removed): `https://arvr.google.com/scene-viewer/1.0?mode=ar_preferred&v=132767069201351460&link=https%3A%2F%2Fprefrontalcortex.de%2F&disable_occlusion=true&file=http%3A%2F%2Famuse.vision%2Far%2FNymph%2FNymph.glb%3Fmode%3Dar_preferred%26v%3D132767069201351460%26sound%3D%26link%3Dhttps%3A%2F%2Fprefrontalcortex.de`

**Comment by hybridherbst** [comment]
Thanks @tpsiaki for letting me know, I had definitely overlooked that.
Is there a way to see debug messags from on-device SceneViewer on Android?

**Comment by tpsiaki** [comment]
You should see some debug messages if you run adb logcat. For example, with your demo this line showed up in logcat:

```Failed to Load Gltf url 'http://amuse.vision/ar/Nymph/Nymph.glb?mode=ar_preferred&v=132767069201351460&sound=&link=https://prefrontalcortex.de' reason: Media Player Error (what 1, extra -2147483648: Error: Unknown Media Error, Reason: System Error)```

Which doesn't exactly say that the sound is the problem, but it was surprising to see that the sound url here was empty and then that the error was about playing media.

Generally if there's an error related to loading the glTF file itself (such as a failure to load an image or something like that) then that will show up in logcat.
